core/jobs/io/gcs_io_test.py

```diff
@@ -0,0 +1,114 @@
+    generation: int = 1
+) -> gcsio_test.FakeFile:
+    """Insert random file into FakeGcsClient.
+
+    Args:
+        client: FakeGcsClient. The fake GCS client for testing purpose.
+        path: str. The file path to where store the content.
+        content: bytes: The content to store.
+        generation: int. The file generation.
+
+    Returns:
+        file: FakeFile. The FakeFile that is stored to FakeGcs.
+    """
+    bucket, name = gcsio.parse_gcs_path(path)
+    file = gcsio_test.FakeFile(bucket, name, content, generation)
+    client.objects.add_file(file)
+    return file
+
+
+class ReadFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.ReadFile."""
+
+    def test_read_from_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        file_name = 'gs://gcsio-test/dummy_file'
+        string = b'testing'
+        insert_random_file(client, file_name, string)
+        filenames = [file_name]
+        filename_p_collec = (
+            self.pipeline
+            |'Create pcoll of filenames' >> beam.Create(filenames)
+            | 'Read file from GCS' >> gcs_io.ReadFile(client)
+        )
+        self.assert_pcoll_equal(filename_p_collec, [string])
+
+
+class WriteFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.WriteFile."""
+
+    def test_write_to_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        file_name_1 = 'gs://gcsio-test/dummy_file_1'
+        file_name_2 = 'gs://gcsio-test/dummy_file_2'
+        string = b'testing'
+        filenames = [
+            {'file': file_name_1, 'data': string},
+            {'file': file_name_2, 'data': string}
+        ]
+        filename_p_collec = (
+            self.pipeline
+            |'Create pcoll of filenames' >> beam.Create(filenames)
+            |'Write to GCS' >> gcs_io.WriteFile(client)
+        )
+        self.assert_pcoll_equal(filename_p_collec, [7, 7])
+
+
+class ReadWriteFileToGCSTest(job_test_utils.PipelinedTestBase):
+    """Read and write file to GCS."""
+
+    def test_read_and_write_file_to_gcs(self) -> None:
+        client = gcsio_test.FakeGcsClient()
+        file_name_1 = 'gs://gcsio-test/dummy_file_1'
+        file_name_2 = 'gs://gcsio-test/dummy_file_2'
+        string = b'testing'
+        filenames = [
+            {'file': file_name_1, 'data': string},
+            {'file': file_name_2, 'data': string}
+        ]
+        filename_p_collec_write = (
+            self.pipeline
+            |'Create pcoll of filenames to write' >> beam.Create(filenames)
+            |'Write to GCS' >> gcs_io.WriteFile(client)
+        )
+        self.assert_pcoll_equal(filename_p_collec_write, [7, 7])
+
+        filename_p_collec_read = (
+            self.pipeline
+            |'Create pcoll of filenames to read' >> beam.Create(filenames)
+            | 'Read file from GCS' >> gcs_io.ReadFile(client)
+        )
+        self.assert_pcoll_equal(filename_p_collec_read, [string, string])
```

Comment:
I was not able to test both the changes together as we cannot use 2 `assert_pcoll_equal` in same function. While testing I got this note -
```
NOTE: This error most likely means you have called more than one PCollection assertion, which is forbidden. This is because running assertions on pipelines require us to wait for it to finish processing all of its data, after which there is nothing left to inspect. If you need to make multiple assertions, then split them into separate test cases.
```

Reply:
Hmm maybe use `JobTestBase` for this class and test it using the job testing functions.